### PR TITLE
support for case of unassigned ipv6 address as chassis ID

### PR DESCRIPTION
**Pull request by mcn29**
The CHASSIS_ID value :: is shown by IOS XE when the remote device uses its IPv6 address as the chassis ID but has no address assigned.

This change prevents the template from throwing an error as soon as it encounters the : character in this peculiar use case

### CODE REVIEW

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.raw [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    S/W revision: SCCP6901.9-3-1-SR2-2
+    Serial number: FCH12345GVM
+    Manufacturer: Cisco Systems, Inc.
+    Model: CP-6901
+    Capabilities: NP, PD, IN
+    Device type: Endpoint Class III
+    Network Policy(Voice): VLAN data, untagged, Layer-2 priority: 5, DSCP: 46
+    Network Policy(Voice Signal): VLAN data, untagged, Layer-2 priority: 3, DSCP: 46
+    PD device, Power source: Unknown, Power Priority: Unknown, Wattage: 2.6
+    Location - not advertised
```
Comment (jmcgill298) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.raw:45]:
```suggestion
    Location - not advertised

```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+PLATFORM: CP-6901
+POWER_CLASS: ''
+POWER_DEVICE_TYPE: ''
+POWER_PAIR: ''
+POWER_PRIORITY: ''
+POWER_REQUESTED: ''
+POWER_SOURCE: ''
+SERIAL: FCH12345GVM
+SW_REVISION: SCCP6901.9-3-1-SR2-2
+VLAN_ID: '999'
```
Comment (jmcgill298) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:23]:
```suggestion
VLAN_ID: '999'

```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+CAPABILITIES: T
```
Comment (jmcgill298) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:2]: ntc_templates will convert these to lowercase, so these will need to be updated to be lowercase.

**Review by jmcgill298 — CHANGES_REQUESTED** [review]
Thanks @mcn29 , just a couple of minor changes

**Comment by mjbear** [comment]
resolves #1884

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+CHASSIS_ID: "::"
+FW_REVISION: 6901.0-0-0-01-05
+HW_REVISION: '1.0'
+LOCAL_INTERFACE: Gi1/0/22
+MAC_ADDRESS: aabb.ccdd.eeff
+MANUFACTURER: Cisco Systems, Inc.
+MGMT_ADDRESS: 10.20.30.40
+NEIGHBOR_DESCRIPTION: Cisco IP Phone 6901,SCCP6901.9-3-1-SR2-2
+NEIGHBOR_INTERFACE: SW PORT
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:11]:
```suggestion
neighbor_interface: "SW PORT"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+NEIGHBOR_NAME: SEPaabbccddeeff.phones.company.domain
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:12]:
```suggestion
neighbor_name: "SEPaabbccddeeff.phones.company.domain"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+NEIGHBOR_PORT_ID: aabbccddeeff:P1
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:13]:
```suggestion
neighbor_port_id: "aabbccddeeff:P1"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+PLATFORM: CP-6901
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:14]:
```suggestion
platform: "CP-6901"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+POWER_CLASS: ''
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:15]:
```suggestion
power_class: ""
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+POWER_DEVICE_TYPE: ''
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:16]:
```suggestion
power_device_type: ""
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 10]
```diff
@@ -0,0 +1,23 @@
+---
+CAPABILITIES: T
+CHASSIS_ID: "::"
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:3]:
```suggestion
chassis_id: "::"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+MANUFACTURER: Cisco Systems, Inc.
+MGMT_ADDRESS: 10.20.30.40
+NEIGHBOR_DESCRIPTION: Cisco IP Phone 6901,SCCP6901.9-3-1-SR2-2
+NEIGHBOR_INTERFACE: SW PORT
+NEIGHBOR_NAME: SEPaabbccddeeff.phones.company.domain
+NEIGHBOR_PORT_ID: aabbccddeeff:P1
+PLATFORM: CP-6901
+POWER_CLASS: ''
+POWER_DEVICE_TYPE: ''
+POWER_PAIR: ''
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:17]:
```suggestion
power_pair: ""
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+POWER_PRIORITY: ''
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:18]:
```suggestion
power_priority: ""
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+POWER_REQUESTED: ''
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:19]:
```suggestion
power_requested: ""
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+POWER_SOURCE: ''
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:20]:
```suggestion
power_source: ""
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+SERIAL: FCH12345GVM
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:21]:
```suggestion
serial: "FCH12345GVM"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+SW_REVISION: SCCP6901.9-3-1-SR2-2
```
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:22]:
```suggestion
sw_revision: "SCCP6901.9-3-1-SR2-2"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml (hunk 2 above)
Comment (mjbear) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:23]:
```suggestion
vlan_id: "999"
```

**Review by mjbear — CHANGES_REQUESTED** [review]

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml
[the remark this answers is not in the document]
Comment (mcn29) [reply tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:1]: Ok, thanks for the suggestions. I'll bear the point in mind in the next PRs.

**Review by jmcgill298 — APPROVED** [review]

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+platform: "CP-6901"
+power_class: ""
+power_device_type: ""
+power_pair: ""
+power_priority: ""
+power_requested: ""
+power_source: ""
+serial: "FCH12345GVM"
+sw_revision: "SCCP6901.9-3-1-SR2-2"
+vlan_id: "999"
```
Comment (jmcgill298) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:23]:
```suggestion
---
parsed_sample:
  capabilities: "T"
  chassis_id: "::"
  fw_revision: "6901.0-0-0-01-05"
  hw_revision: "1.0"
  local_interface: "Gi1/0/22"
  mac_address: "aabb.ccdd.eeff"
  manufacturer: "Cisco Systems, Inc."
  mgmt_address: "10.20.30.40"
  neighbor_description: "Cisco IP Phone 6901,SCCP6901.9-3-1-SR2-2"
  neighbor_interface: "SW PORT"
  neighbor_name: "SEPaabbccddeeff.phones.company.domain"
  neighbor_port_id: "aabbccddeeff:P1"
  platform: "CP-6901"
  power_class: ""
  power_device_type: ""
  power_pair: ""
  power_priority: ""
  power_requested: ""
  power_source: ""
  serial: "FCH12345GVM"
  sw_revision: "SCCP6901.9-3-1-SR2-2"
  vlan_id: "999"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+  platform: "CP-6901"
+  power_class: ""
+  power_device_type: ""
+  power_pair: ""
+  power_priority: ""
+  power_requested: ""
+  power_source: ""
+  serial: "FCH12345GVM"
+  sw_revision: "SCCP6901.9-3-1-SR2-2"
+  vlan_id: "999"
```
Comment (jmcgill298) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:24]:
```suggestion
  - capabilities: "T"
    chassis_id: "::"
    fw_revision: "6901.0-0-0-01-05"
    hw_revision: "1.0"
    local_interface: "Gi1/0/22"
   mac_address: "aabb.ccdd.eeff"
    manufacturer: "Cisco Systems, Inc."
    mgmt_address: "10.20.30.40"
    neighbor_description: "Cisco IP Phone 6901,SCCP6901.9-3-1-SR2-2"
    neighbor_interface: "SW PORT"
    neighbor_name: "SEPaabbccddeeff.phones.company.domain"
    neighbor_port_id: "aabbccddeeff:P1"
    platform: "CP-6901"
    power_class: ""
    power_device_type: ""
    power_pair: ""
    power_priority: ""
    power_requested: ""
    power_source: ""
    serial: "FCH12345GVM"
    sw_revision: "SCCP6901.9-3-1-SR2-2"
    vlan_id: "999"
```

File: tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml [hunk 19]
```diff
@@ -0,0 +1,24 @@
+---
+parsed_sample:
+  - capabilities: "T"
+    chassis_id: "::"
+    fw_revision: "6901.0-0-0-01-05"
+    hw_revision: "1.0"
+    local_interface: "Gi1/0/22"
+   mac_address: "aabb.ccdd.eeff"
+    manufacturer: "Cisco Systems, Inc."
+    mgmt_address: "10.20.30.40"
```
Comment (jmcgill298) [inline tests/cisco_ios/show_lldp_neighbors_detail/cisco_ios_show_lldp_neighbors_detail6.yml:10]:
```suggestion
    mac_address: "aabb.ccdd.eeff"
    manufacturer: "Cisco Systems, Inc."
    mgmt_address: "10.20.30.40"
```

**Comment by mjbear** [comment]
Thank you @jmcgill298! :fireworks:

**Pull request merged**